freqtrade/strategy/strategyupdater.py

```diff
@@ -0,0 +1,260 @@
+import ast
+import os
+import shutil
+from pathlib import Path
+
+import astor
+
+
+class StrategyUpdater:
+    name_mapping = {
+        'ticker_interval': 'timeframe',
+        'buy': 'enter_long',
+        'sell': 'exit_long',
+        'buy_tag': 'enter_tag',
+        'sell_reason': 'exit_reason',
+
+        'sell_signal': 'exit_signal',
+        'custom_sell': 'custom_exit',
+        'force_sell': 'force_exit',
+        'emergency_sell': 'emergency_exit',
+
+        # Strategy/config settings:
+        'use_sell_signal': 'use_exit_signal',
+        'sell_profit_only': 'exit_profit_only',
+        'sell_profit_offset': 'exit_profit_offset',
+        'ignore_roi_if_buy_signal': 'ignore_roi_if_entry_signal',
+        'forcebuy_enable': 'force_entry_enable',
+    }
+
+    function_mapping = {
+        'populate_buy_trend': 'populate_entry_trend',
+        'populate_sell_trend': 'populate_exit_trend',
+        'custom_sell': 'custom_exit',
+        'check_buy_timeout': 'check_entry_timeout',
+        'check_sell_timeout': 'check_exit_timeout',
+        # '': '',
+    }
+    # order_time_in_force, order_types, unfilledtimeout
+    otif_ot_unfilledtimeout = {
+        'buy': 'entry',
+        'sell': 'exit',
+    }
+
+    # create a dictionary that maps the old column names to the new ones
+    rename_dict = {'buy': 'enter_long', 'sell': 'exit_long', 'buy_tag': 'enter_tag'}
+
+    def start(self, config, strategy_obj: dict) -> None:
+        """
+        Run strategy updater
+        It updates a strategy to v3 with the help of the ast-module
+        :return: None
+        """
+
+        source_file = strategy_obj['location']
+        print(f"started conversion of {source_file}")
+        strategies_backup_folder = Path.joinpath(config['user_data_dir'], "strategies_orig_updater")
+        target_file = Path.joinpath(strategies_backup_folder, strategy_obj['location_rel'])
+
+        # read the file
+        with open(source_file, 'r') as f:
+            old_code = f.read()
+        if not os.path.exists(strategies_backup_folder):
+            os.makedirs(strategies_backup_folder)
+
+        # backup original
+        # => currently no date after the filename,
+        # could get overridden pretty fast if this is fired twice!
+        # The folder is always the same and the file name too (currently).
+        shutil.copy(source_file, target_file)
+
+        # update the code
+        new_code = StrategyUpdater.update_code(self, old_code)
```

Comment:
i'm surprised this does even work.

update_code is not a static method (and we're in an instance method anyway) - so the correct way to call this would be 
`new_code = self.update_code(old_code)`